### PR TITLE
chore: fix `version` script

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Follow-up of #4849

### CODE REVIEW

File: scripts/utils.js [hunk 1]
```diff
@@ -101,7 +101,11 @@ async function updateRemixVersion(packageName, nextVersion, successMessage) {
         config.devDependencies[`@remix-run/${pkg}`] = nextVersion;
       }
       if (config.peerDependencies?.[`@remix-run/${pkg}`]) {
-        config.peerDependencies[`@remix-run/${pkg}`] = nextVersion;
+        let isRelaxedPeerDep =
+          config.peerDependencies[`@remix-run/${pkg}`]?.startsWith("^");
+        config.peerDependencies[`@remix-run/${pkg}`] = `${
+          isRelaxedPeerDep ? "^" : ""
+        }${nextVersion}`;
```
Comment (MichaelDeBoey) [inline scripts/utils.js:108]: @chaance loosened this in #4736 for `@remix-run/dev`

File: scripts/utils.js [hunk 2]
```diff
@@ -165,7 +169,8 @@ const updateDenoImportMap = async (importMapPath, nextVersion) => {
       let [packageName, importPath] =
         getPackageNameFromImportSpecifier(importName);
 
-      return remixPackagesFull.includes(packageName)
+      return remixPackagesFull.includes(packageName) &&
+        importName !== "@remix-run/deno"
```
Comment (MichaelDeBoey) [inline scripts/utils.js:173]: `@remix-run/deno` code is already a Deno module, so just get types for it directly from `node_modules/`

https://github.com/remix-run/remix/blob/d5ca496c47ff84c02509b0957e54869de014521e/templates/deno/.vscode/resolve_npm_imports.json#L10-L11

File: scripts/utils.js (hunk 1 above)
Comment (mcansh) [reply scripts/utils.js:115]: i'll defer to @chaance on this :)

File: scripts/utils.js [hunk 3]
```diff
@@ -213,11 +218,6 @@ async function incrementRemixVersion(nextVersion) {
 
   // Update deployment script `@remix-run/dev` version
   await updateDeploymentScriptVersion(nextVersion);
-
-  // Commit and tag
-  execSync(`git commit --all --message="Version ${nextVersion}"`);
```
Comment (chaance) [inline scripts/utils.js:218]: Why are we removing this?

Comment (MichaelDeBoey) [reply scripts/utils.js:218]: We're using Changeset for creating versions, not a manual commit

File: scripts/utils.js (hunk 1 above)
Comment (mcansh) [reply scripts/utils.js:115]: we still manually commit for nightly and experimental releases

File: scripts/utils.js (hunk 3 above)
Comment (mcansh) [reply scripts/utils.js:218]: we still manually commit for nightly and experimental releases

**Review by chaance — APPROVED** [review]

**Pull request merged**